extensions/numi/src/services/checkinstall.ts

```diff
@@ -10,21 +15,54 @@ async function isNumiInstalled() {
   );
 }
 
-export async function checkNumiInstallation() {
-  if (!(await isNumiInstalled())) {
-    const options: Toast.Options = {
-      style: Toast.Style.Failure,
-      title: "Numi is not installed.",
-      message: "Install it from: https://numi.app/",
-      primaryAction: {
-        title: "Go to https://numi.app/",
-        onAction: (toast) => {
-          open("https://numi.app/");
-          toast.hide();
+export async function checkNumiInstallation(cli: boolean) {
+  if (!cli) {
+    if (!(await isNumiInstalled())) {
+      const options: Toast.Options = {
+        style: Toast.Style.Failure,
+        title: "Numi is not installed.",
+        message: "Install it from: https://numi.app/",
+        primaryAction: {
+          title: "Go to https://numi.app/",
+          onAction: (toast) => {
+            open("https://numi.app/");
+            toast.hide();
+          },
+        },
+      };
+
+      await showToast(options);
+    }
+  } else {
+    if (!(await isNumiCliInstalled())) {
+      const options: Toast.Options = {
+        style: Toast.Style.Failure,
+        title: "Numi CLI is not installed.",
+        message: "Install it from: https://github.com/nikolaeu/numi#numi-cli",
+        primaryAction: {
+          title: "Go to https://github.com/nikolaeu/numi#numi-cli",
+          onAction: (toast) => {
+            open("https://github.com/nikolaeu/numi#numi-cli");
+            toast.hide();
+          },
         },
-      },
-    };
+      };
+
+      await showToast(options);
+    }
+  }
+}
 
-    await showToast(options);
+export async function isNumiCliInstalled(path?: string) {
+  try {
+    const res = await execp(`${path || DefaultPreferences.numi_cli_binary_path} --version`, { shell: "/bin/bash" });
```

Comment:
Always use `DefaultPreferences.numi_cli_binary_path`; if there is any security flaw that permits the call of this function, modifying the path parameter could handle a big problem if the user has failures. They need to set up the correct path on the preferences.

Reply:
I provide the path as an optional parameter if the user installs it outside of the options recommended by Numi-Cli, at this case using the homebrew bin path. is it better to use this default binary path without providing an option for the user to change it?

Reply:
Ok, I got it. Always use the **numi_cli_binary_path** from the preferences.

`extensions/numi/src/types`
```ts

export interface Preferences {
  max_history_elemets: string;
  use_numi_cli: boolean;
  numi_cli_binary_path: string;
}

```
At `extensions/numi/src/services/checkinstall.ts`
```ts
import { getPreferenceValues } from "@raycast/api";
import { Preferences } from "../types/index";
...

const {  numi_cli_binary_path } = getPreferenceValues<Preferences>();

export async function isNumiCliInstalled(path?: string) {
  if (path && path.trim() === "") path = undefined;
  try {
    const res = await execp(`${numi_cli_binary_path} --version`, { shell: "/bin/bash" });

   if (res.stderr) return false;
    if (!res.stdout.includes("numi-cli")) return false;
    return res.stdout;
  } catch (error) {
    return false;
  }
}

```

At `extensions/numi/src/services/requests.tsx`
```ts
import { getPreferenceValues } from "@raycast/api";
import { Preferences } from "../types/index";

const {  numi_cli_binary_path } = getPreferenceValues<Preferences>();

export const queryWithNumiCli = async (q?: string, path?: string): Promise<string[]> => {
  q = q?.replace(/"/g, '\\"') || ""; // safe " in query
  const command = `${numi_cli_binary_path} "${q}"`;

  const res = await execp(command, { shell: "/bin/bash" });
  if (res.stderr) {
    console.error(res.stderr);
    return [];
  }
  let result = res.stdout;
  if (result.endsWith("\n")) result = result.slice(0, -1);
  result = Buffer.from(result).toString("utf8");
  return [result];
};

```

Reply:
yes sure, I use the preferences of the user via the argument parameter which is passed via the function `isNumiCliIstalled` and function `queryWithNumiCli` parameters (path?: string).

in current code, i use `numi_cli_binary_path` from user prefs and if user not provided (preference undefined/null) so use a default value from `DefaultPreferences.numi_cli_binary_path`

like 
```ts
const {  numi_cli_binary_path } = getPreferenceValues<Preferences>();
const command = `${numi_cli_binary_path || DefaultPreferences.numi_cli_binary_path} "${q}"`;
```

or you want to me keep use from user pref only? without default value if user not setted? 
```ts
const {  numi_cli_binary_path } = getPreferenceValues<Preferences>();
const command = `${numi_cli_binary_path} "${q}"`;
```

Reply:
Yep, always use the preference bin path.

Reply:
sure, changed on https://github.com/raycast/extensions/pull/7412/commits/d3b9beb95d0b16d7031726c9026db5e2373b8301